engine/src/main/java/nl/inl/blacklab/resultproperty/PropertyValueDoc.java

```diff
@@ -1,68 +1,64 @@
 package nl.inl.blacklab.resultproperty;
 
-import org.apache.lucene.document.Document;
-
 import nl.inl.blacklab.search.BlackLabIndex;
-import nl.inl.blacklab.search.Doc;
 
 /** Property value that represents a BlackLab document */
 public class PropertyValueDoc extends PropertyValue {
-    private Doc value;
+    private BlackLabIndex index;
+
+    private int docId;
 
     @Override
-    public Doc value() {
-        return value;
+    public Integer value() {
+        return docId;
     }
     
-    public int id() {
-        return value.id();
-    }
-    
-    public Document luceneDoc() {
-        return value.luceneDoc();
-    }
+    /*public Document luceneDoc() {
+        return index.luceneDoc(docId);
+    }*/
 
-    public PropertyValueDoc(Doc doc) {
-        this.value = doc;
+    public PropertyValueDoc(BlackLabIndex index, int id) {
+        this.index = index;
+        this.docId = id;
     }
 
     @Override
     public int compareTo(Object o) {
-        return value.id() - ((PropertyValueDoc) o).value.id();
+        return docId - ((PropertyValueDoc) o).docId;
     }
 
     @Override
     public int hashCode() {
-        return value.hashCode();
+        return Integer.hashCode(docId);
     }
 
     @Override
     public boolean equals(Object obj) {
         if (obj == this)
             return true;
         if (obj instanceof PropertyValueDoc) {
-            return value.id() == ((PropertyValueDoc) obj).value.id();
+            return docId == ((PropertyValueDoc) obj).docId;
         }
         return false;
     }
 
     public static PropertyValue deserialize(BlackLabIndex index, String info) {
-        Doc v;
+        int id;
         try {
-            v = index.doc(Integer.parseInt(info));
+            id = Integer.parseInt(info);
         } catch (NumberFormatException e) {
-            v = null;
+            id = -1;
```

Comment:
Perhaps writting warning about this event would aid in debugging an issue involving this code path.

Reply:
Agreed, I added a warning for this in a few places. I do want to prevent is log chatter from messy (i.e. expected) input data, but this is about query parameters sent to BLS, which probably usually conform to the expected type.